R/resource_update.R

```diff
@@ -102,19 +110,27 @@
 #' (xxx <- resource_update(xx, path=newpath))
 #' browseURL(xxx$url)
 #' }
-resource_update <- function(id, path, extras = list(),
+resource_update <- function(id, path = NULL, extras = list(),
   url = get_default_url(), key = get_default_key(),
   as = 'list', ...) {
 
   assert(extras, "list")
   id <- as.ckan_resource(id, url = url)
-  path <- path.expand(path)
-  up <- upfile(path)
-  format <- pick_type(up$type)
-  body <- list(id = id$id, format = format, upload = up,
-    last_modified =
-      format(Sys.time(), tz = "UTC", format = "%Y-%m-%d %H:%M:%OS6"),
-    url = "update")
+  if (is.character(path)) {
```

Comment:
only gripe here is that some of these elements that go in to `body` are the same for both if/else paths: `id`, `last_modified`, `url` - maybe those could go below the if/else block to reduce duplicated code lines?

Reply:
ah thanks, also made me verify that `format` isn't dropped and doesn't beed to be included when `path` is omitted